### ISSUE TITLE
Accept Multiple Connections on server side

### ISSUE THREAD

**Issue by git4unrealnondev** [kotauskas/interprocess#66]
No idea how feasible it would be to have one server respond to multiple requests at the same time but it is a feature that I would like.
This is because I'm trying to implement a plugins system in an application.
The issue is that if a process takes 1-2 seconds to run then nothing else can call into the plugin manager while that happens while it waits for the interprocess server to respond.
I currently use the local_sockets process for it. Have yet to try tokio

If this isn't feasible without a major rewrite / isn't what this crate was meant to do then I understand.

**Describe the solution you'd like**
Ideally a way for one server to respond to multiple connections at the same time.

**Describe alternatives you've considered**
I've tried spooling multiple servers but that increases code complexity and I'm more likely to make a mistake.

**Comment by kotauskas** [comment]
This feature has always been part of the non-Tokio part of Interprocess the same way the ability to handle concurrent clients has always been in the standard library's `TcpListener`. [`Listener::accept()`](<https://docs.rs/interprocess/2.1.1/interprocess/local_socket/traits/trait.Listener.html#tymethod.accept>) takes `&self`, meaning that a `Listener` can be shared across threads. By spawning some fixed amount of threads (and possibly more as needed) and calling `.accept()` in every thread, you can get your OS to automatically dispatch clients to those threads for you.

The choice between using a thread pool-based blocking server and Tokio depends primarily on how lightweight you want your server to be. Tokio with the single-thread runtime can be quite lightweight, but very low-footprint scenarios might require you to resort to the non-Tokio side of things.

The bottom line: you should just use asynchronous I/O (which exists to tackle your specific problem and many more) with the help of Tokio unless you have good reasons not to.
